frontend/src/features/map_planning/components/timeline/ItemSliderPicker.tsx

```diff
@@ -0,0 +1,307 @@
+import './styles/timeline.css';
+import React, { useRef, useState, useEffect, ReactNode, useCallback } from 'react';
+
+export type ItemSliderItem = {
+  key: number;
+  content: ReactNode;
+};
+
+interface ItemSliderPickerProps {
+  /** List of items to display */
+  items: ItemSliderItem[];
+  /** Callback when an item is selected */
+  onChange: (selectedItemKey: number) => void;
+  /** Callback that fires when the left end of the slider is reached */
+  leftEndReached?: () => void;
+  /** Callback that fires when the right end of the slider is reached */
+  rightEndReached?: () => void;
+  /** initial value */
+  value?: number;
+  /** id for testing */
+  dataTestId?: string;
+}
+
+const ItemSliderPicker: React.FC<ItemSliderPickerProps> = ({
+  items,
+  onChange,
+  value,
+  leftEndReached,
+  rightEndReached,
+  dataTestId,
+}: ItemSliderPickerProps) => {
+  const [selectedItemKey, setSelectedItemKey] = useState<number>(0);
+
+  const sliderContainerRef = useRef<HTMLDivElement | null>(null);
+  const isDragging = useRef<boolean>(false);
+  const dragStartX = useRef<number | null>(null);
+
+  const scrollPositionOnMouseDown = useRef<number | null>(null);
+  const isScrollingToIndex = useRef<boolean>(false);
+
+  const detectAndNotifyIfBordersIsReached = () => {
+    const container = sliderContainerRef.current;
+    if (container) {
```

Comment:
```tsx
if (container) {
```
to
```tsx
if (!container) return;
```
I generally prefer to invert the if conditions as guard clauses.
Makes for more readable code imo.
Especially if there are nested ifs inside.